src/memory/memory.cc

```diff
@@ -1,3 +1,48 @@
 #include "memory/memory.hh"
 
-namespace sim {}
+namespace sim {
+
+Memory::Memory() : mem{}, stats{} {}
+
+Word Memory::loadWord(Addr addr) {
+
+  stats.numLoads++;
+  Word word = kDummyWord;
+
+  auto it = mem.find(addr);
+  if (it != mem.end())
+    word = it->second;
+  else
+    word = pageFaultHandle(addr);
+
+  return word;
+}
+
+void Memory::storeWord(Addr addr, Word word) {
+
+  stats.numStores++;
+  mem[addr] = word;
+}
+
+Word Memory::pageFaultHandle(Addr addr) {
+
+  (void)addr;
+  stats.numPageFaults++;
+  std::cout << "Placeholder for page fault" << std::endl;
+  return kDummyWord;
+}
+
+std::size_t Memory::getCurrMemSize() const { return mem.size(); }
+
+void Memory::printMemStats(std::ostream &ost) const {
+
+  ost << "Memory stats:" << std::endl;
+  ost << "Loads: " << stats.numLoads << std::endl;
+  ost << "Stores: " << stats.numStores << std::endl;
+  ost << "PageFaults: " << stats.numPageFaults << std::endl;
+  ost << "End of memory stats." << std::endl;
+}
+
+const Memory::MemoryStats &Memory::getMemStats() const { return stats; }
+
+} // namespace sim
```

Comment:
nit: no newline at end of file. To avoid this you can add 
`{
    "editor.renderWhitespace": "all",
    "files.trimTrailingWhitespace": true,
    "files.trimFinalNewlines": true,
    "files.insertFinalNewline": true,
}
` 
to settings.json in vscode. The same in memory.test.cc and common.hh

Reply:
Thanks, reworked in  ([2d0e439](https://github.com/106-inc/sim2022/pull/15/commits/2d0e439e617cd8b16865f4aaf87cdf1305cb98db))

Reply:
Still relevant

Reply:
Still relevant